core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1119,6 +1119,15 @@ class ReplicaManager(val config: KafkaConfig,
         else
           Set.empty[Partition]
 
+        /*
+         * KAFKA-8392
+         * For topic partitions of which the broker is no longer a leader, delete metrics (bytesIn, bytesOut, messagesIn)
```

Comment:
Similarly, if a broker is not the leader for any partition in a topic, there is no need to keep the ProduceMessageConversionsPerSec and ReplicationBytesOutPerSec metric. If a broker is not the follower for any partition, there is no need to keep the ReplicationBytesInPerSec metric.

Reply:
Thanks! I've added a new method to remove metrics not needed for a leader-only broker: https://github.com/apache/kafka/pull/6977/files#diff-449a86e6eeefa27f776bb8dab30f6b2eR247